src/components/Switch.tsx

```diff
@@ -23,31 +24,33 @@ export function Switch({
   id: string;
   /** The label for the switch */
   label: string;
+  /** Whether to autofocus this input */
+  autoFocus?: true;
 }): JSX.Element {
-  const handleClick = (
-    event: JSX.TargetedEvent<HTMLInputElement, Event>,
-  ): void => {
-    event.preventDefault();
-    handleSwitch(!checked, event);
-  };
+  const handleClick = (e: JSX.TargetedEvent<HTMLInputElement, Event>): void =>
+    handleSwitch(!checked, e);
 
   return (
-    <Fragment>
-      <label className="switch label">
-        <input
-          key={id}
-          className="switch switch-checkbox screen-reader"
-          id={`switch-${id}`}
-          type="checkbox"
-          disabled={disabled}
-          checked={checked}
-          onClick={handleClick}
-        />
-        <span className="switch switch-background">
-          <span className="switch switch-button" />
-        </span>
-        {label}
-      </label>
-    </Fragment>
+    <label className="switch label">
+      <input
+        className="switch switch-checkbox screen-reader"
+        id={`switch-${id}`}
+        type="checkbox"
+        disabled={disabled}
+        checked={checked}
+        onClick={handleClick}
+        onKeyPress={(e) => {
+          if (e.key !== 'Enter') return;
+          handleSwitch(!checked, e);
+        }}
```

Comment:
I would tend to leave this without custom handling. SRs will know this is a checkbox and are used to interacting with checkboxes via the keyboard. I believe typical controls for checkboxes are to use the spacebar to check/uncheck it.

Reply:
i dont really seem the harm here since spacebar still works fine, this just makes us capture enter presses as usable input also?

Reply:
Oh, I see. I though this was disabling all other key presses. "Enter" in forms is usually for the triggering submit, but if this isn't in a form, then it probably doesn't matter too much.